tests/jeos/check_efi_setup.pm

```diff
@@ -0,0 +1,230 @@
+# SUSE's openQA tests
+#
+# Copyright © 2020 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+#
+# Summary: Check EFI boot in JeOS images
+#   1) Get gpt drive and find ESP
+#   2) Check ESP partition alignment
+#   3) Perform fsck on ESP
+#   4) Check EFI settings after firstboot
+#   5) Update grub.cfg and reboot
+#   6) Check EFI settings after grub.cfg update
+#   7) Reinstall shim and reboot
+#   8) Check EFI settings after shim reinstal
+#   9) Reinitialize bootloader using pbl
+#   10) Check EFI settings after reinit
+# Maintainer: Martin Loviska <mloviska@suse.com>
+
+use strict;
+use warnings;
+use base "opensusebasetest";
+use testapi;
+use utils 'zypper_call';
+use version_utils qw(is_leap is_opensuse);
+use jeos;
+use constant {
+    SYSFS_EFI_BITS      => '/sys/firmware/efi/fw_platform_size',
+    FSTAB               => '/etc/fstab',
+    GRUB_DEFAULT        => '/etc/default/grub',
+    GRUB_CFG            => '/boot/grub2/grub.cfg',
+    SYSCONFIG_BOOTLADER => '/etc/sysconfig/bootloader'
+};
+
+sub efibootmgr_current_boot {
+    # helper function to retrieve location of efi executable
+    my $get_efi_exec = sub {
+        my $str = shift;
+        $str or die "Empty record in efibootmgr parser!\n";
+        $str =~ s/\\/\//g;
+        return substr $str, index($str, '(') + 1, (index($str, ')') - 1) - index($str, '(');
+    };
+
+    my @ebm_raw = split /\n/, script_output 'efibootmgr --verbose';
+    die "EFI variables are not supported\n" if ($ebm_raw[0] =~ /EFI variables are not supported/);
```

Comment:
isn't `efibootmgr` will return non-zero error code so `script_output` would die even before this line ?

Reply:
Good point, I had a feeling that it was not reporting rcs somewhere. Maybe mistaking it with some other util. let me check

Reply:
Yes, completely right
https://github.com/rhboot/efibootmgr/blob/master/src/efibootmgr.c#L1764